### DISCUSSION TITLE
footpint plotting

### DISCUSSION THREAD

**Discussion by Dazcam** [GreenleafLab/ArchR#1107] [Questions / Documentation]
Is it possible to retrieve/access the ggplot2 object that is produced when `plotFootprints()` is run? At present plots are produced as pdfs and when multiple motifs are queried these are aggregated in one plot.  

I'd like to manipulate the the aesthetics of these plots manually and have tinkered with the `.ggFootprint` function (see below) but can't easily pull out the ggplot2 object without getting lost in the weeds of your (excellent) code.

```R
ArchR:::.ggFootprint(seFoot = seFoot, 
                     name = 'DLX5_412',
                     smoothWindow = 5,
                     normMethod = "Subtract",
                     flank = 250,
                     flankNorm = 50,
                     pal = NULL,
                     baseSize = 6,
                     logFile = NULL
                     )
```

Many Thanks.

**Comment by rcorces** [comment]
see the `plot` parameter for `plotFootprints()`:

<html>
<body>


plot | A boolean value indicating whether or not the footprints should be plotted (TRUE) or returned as grob objects (FALSE).
-- | --



</body>
</html>

**Reply by Dazcam** [reply]
Many thanks for responding. 

I saw that a grob was available and have tried to tinker with the grob object to try to alter the legend, text size of the axes labels (which are both tiny!) and the default colors assigned to the cell types. Tbh, I've never worked with grob objects directly so my strategy for this was to change the object back to a `ggplot2` object and use the standard `ggplot2` commands to alter these things, for example:

```R
fig5G <- ggplotify::as.ggplot(fig_5G_grob[[NEUROD2_73]])

fig5G +
  gg_title(NULL) +
  theme(legend.position = "none",
        legend.title=element_blank(),
        plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm"),
        axis.title.x = element_text(colour = "#000000", size = 14),
        axis.title.y = element_text(colour = "#000000", size = 14),
        axis.text.x  = element_text(colour = "#000000", size = 12, vjust = 0.5),
        axis.text.y  = element_text(colour = "#000000", size = 12),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank())
```

I guess the issue is that the grob is a composite of multiple plots. I [see](https://github.com/GreenleafLab/ArchR/blob/6765ad962d4d8dcb292a326071c9b5c30c25918e/R/Footprinting.R#L571) you use `ggplot2` to construct these plots then align them with `ggAlignPlots`. Is there any way to access/extract the `plotFootDF` and `plotFootBiasDF` data that you use to construct these plots? If so, I could then those with your `ggplot2` code to manipulate these plots myself? I don't see a way to pull these data from the grob directly (or indirectly after using `as.ggplot`).

**Reply by rcorces** [reply]
This isnt a use case that we can support. I think if you want these objects, you should create your own copy of the `plotFootprints()` function and have it return what you want.

**Reply by Dazcam** [reply]
Thank you. I understand. But I do think it is worth considering making it simpler for users to access data used for plotting, across all ArchR functions, directly. You do provide access to plotting data for other functions i.e. the motif matrix used to plot motif enrichment.
